### ISSUE TITLE
Cookbook should use rubygems_url client config for rubysource attribute.

### ISSUE THREAD

**Issue by donwlewis** [chef-cookbooks/lvm#144]
### Cookbook version
4.1.12

### Chef-client version
13.6

### Platform Details
RHEL7

### Scenario:
Cookbook should honor honor the `rubygems_url` in the chef client config file for air gapped environments.

### Steps to Reproduce:
Run cookbook in air gapped environment without specifying `default['lvm']['rubysource']`, and having `rubygems_url` in the client configuration.

### Expected Result:
Client should use client configured rubygems URL.

### Actual Result:
Client does not and fails in air gapped environments.

**Comment by ChrisAWalker** [comment]
Why don't you just override that attribute and redirect it to your internal gem source?  This is a normal activity as part of running chef.  (proven to work in my company where we are 'air gapped'.

**Comment by donwlewis** [comment]
That is what I do, but feel like the client config should be honored.  Why not use what the client is configured for?  If not, why even configure the client?

**Comment by bby-bishopclark** [comment]
Indeed, why even set #{chef_etc_dir}/gemrc, /root/.gemrc, /root/.bundle/config and probably one more when we can have a new time-bomb with every new cookbook we consider importing?

This cookbook needs a break, really, as it's created a lot of heart-ache with the rubygems version lock-step, and it'd be great if this wasn't another shot in the foot.  Can we consider having it look for **another** system-wide value as well?  

    default['lvm']['rubysource'] ||= node.read('rubysource') || 'https://rubygems.org'

I know, I know, xkcd.com/927 and all.  But if it listens to a centralized value, it may be an improvement on the way to actually using the config in the client.
